### PR TITLE
Build RAJA-Examples as a subdirectory

### PR DESCRIPTION
**Pull request by davidbeckingsale**
This PR lets you optionally build the RAJA-examples repository as part of RAJA.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -97,3 +97,11 @@ endif()
 if(RAJA_ENABLE_EXAMPLES)
   add_subdirectory(examples)
 endif()
+
+if(RAJA_ENABLE_APPLICATIONS)
+  if (NOT EXISTS ${PROJECT_SOURCE_DIR}/applications/CMakeLists.txt)
+    message(STATUS "Cloning RAJA example applications...")
+    execute_process(COMMAND git clone https://github.com/LLNL/RAJA-examples.git ${PROJECT_SOURCE_DIR}/applications)
```
Comment (willkill07) [inline CMakeLists.txt]: Do we want to leave the potential for additional repos to be incorporated? If so, perhaps the RAJA-examples checkout should live as a subdirectory within the applications directory. Else, 👍

**Comment by DavidPoliakoff** [comment]
Ditto Will's comment, applications/llnl-proxies might be the subdir. Also, 👍

**Comment by davidbeckingsale** [comment]
I agree, perhaps a better structure would work if we would be incorporating things like the performance suite or other proxies. Maybe an `extra` directory with subdirectories for each additional repo?

**Comment by willkill07** [comment]
I like `extra` with the repo name directly under

**Comment by davidbeckingsale** [comment]
I've changed the structure,  and as @DavidPoliakoff suggested, called the applications repository `llnl-raja-proxies`

If this is the final directory structure, I'll rename the RAJA-examples repository also.

**Comment by willkill07** [comment]
LGTM

**Comment by rhornung67** [comment]
@willkill07 you're making me feel old. I had to Google your last comment.

@davidbeckingsale :+1:

**Pull request merged**